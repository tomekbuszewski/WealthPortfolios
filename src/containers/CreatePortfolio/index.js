import * as React from 'react';

import Communication from '@src/services/Communication';
import { ENDPOINTS } from '@config';

import View from '@src/views/CreatePortfolio';

class CreatePortfolio extends React.Component {
  constructor(props) {
    super(props);

    const id = Math.round(Math.random() * (100 - 10) + 10);
    this.state = {
      data: {
        id, // This should not be required
      },
      success: null,
    };
  }

  setValue = (event) => {
    const { name, value } = event.target;
    const { data } = this.state;

    this.setState({
      data: {
        ...data,
        [name]: value,
      },
    });
  }

  submit = (e) => {
    e.preventDefault();

    this.create();
  }

  resetStatus = () => {
    setTimeout(() => {
      this.setState({ success: null });
    }, 2000);
  }

  create() {
    const { data } = this.state;

    const create = new Communication({
      url: ENDPOINTS.PORTFOLIOS,
      method: 'post',
      data,
    });

    create()
      .then(({ status }) => this.setState({ success: status === 200 }, this.resetStatus))
      .catch(() => this.setState({ success: false }, this.resetStatus));
  }

  render() {
    const { data, success } = this.state;

    return (
      <View
        data={data}
        onChange={this.setValue}
        submit={this.submit}
        success={success}
      />
    );
  }
}

export default CreatePortfolio;
