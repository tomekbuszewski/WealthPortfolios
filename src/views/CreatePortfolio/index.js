import * as React from 'react';
import Props from 'prop-types';

import Input from '@src/components/Input';
import Section from '@src/components/Section';
import Notify from '@src/components/Notification';

import FormController from '@src/containers/FormController';

const CreatePortfolio = (props) => {
  const {
    onChange,
    data,
    submit,
    success,
  } = props;

  return (
    <Section title="Create portfolio">
      <FormController>
        {success === true && <Notify>Creation successful</Notify>}
        {success === false && <Notify type="warning">Creation failed</Notify>}
        <Input label="Portfolio name" name="name" onChange={onChange} value={data.name ? data.name : ''} />
        <Input label="Client name" name="clientName" onChange={onChange} value={data.clientName ? data.clientName : ''} />
        <Input label="Initial wallet value" name="value" onChange={onChange} value={data.value ? data.value : ''} />
        <Input label="Wallet currency" name="currency" onChange={onChange} value={data.currency ? data.currency : ''} />
        <button type="submit" onClick={submit} className="button is-primary">Submit</button>
      </FormController>
    </Section>
  );
};

CreatePortfolio.propTypes = {
  onChange: Props.func.isRequired,
  submit: Props.func.isRequired,
  data: Props.objectOf(Props.oneOfType([Props.string, Props.number])).isRequired,
  success: Props.bool,
};

CreatePortfolio.defaultProps = {
  success: null,
};

export default CreatePortfolio;
