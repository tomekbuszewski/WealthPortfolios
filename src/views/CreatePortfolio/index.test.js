import * as React from 'react';
import renderer from 'react-test-renderer';

import CreatePortfolio from './';

const dataEmpty = {};
import data from '@mocks/portfolios';

describe ('CreatePortfolio tests', () => {
  it ('should match snapshot with empty data', () => {
    const component = renderer.create(<CreatePortfolio
      onChange={() => false}
      submit={() => false}
      data={dataEmpty}
    />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it ('should match snapshot with success true', () => {
    const component = renderer.create(<CreatePortfolio
      onChange={() => false}
      submit={() => false}
      data={dataEmpty}
      success={true}
    />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it ('should match snapshot with success false', () => {
    const component = renderer.create(<CreatePortfolio
      onChange={() => false}
      submit={() => false}
      data={dataEmpty}
      success={false}
    />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it ('should match snapshot with provided data', () => {
    const component = renderer.create(<CreatePortfolio
      onChange={() => false}
      submit={() => false}
      data={data[0]}
    />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
