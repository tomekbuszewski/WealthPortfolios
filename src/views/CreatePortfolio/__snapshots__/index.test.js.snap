// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePortfolio tests should match snapshot with empty data 1`] = `
<section
  className="sc-bdVaJa gPOcxX section hero"
>
  <header
    className="title"
  >
    Create portfolio
  </header>
  <div
    className="content is-large"
  >
    <form>
      <label
        className="label"
        htmlFor="name"
      >
        Portfolio name
        <input
          className="input"
          disabled={false}
          id="name"
          name="name"
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <label
        className="label"
        htmlFor="clientName"
      >
        Client name
        <input
          className="input"
          disabled={false}
          id="clientName"
          name="clientName"
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <label
        className="label"
        htmlFor="value"
      >
        Initial wallet value
        <input
          className="input"
          disabled={false}
          id="value"
          name="value"
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <label
        className="label"
        htmlFor="currency"
      >
        Wallet currency
        <input
          className="input"
          disabled={false}
          id="currency"
          name="currency"
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <p
        className="notification is-primary subtitle is-6"
      >
        Please fill all the fields to submit form
      </p>
    </form>
  </div>
</section>
`;

exports[`CreatePortfolio tests should match snapshot with provided data 1`] = `
<section
  className="sc-bdVaJa gPOcxX section hero"
>
  <header
    className="title"
  >
    Create portfolio
  </header>
  <div
    className="content is-large"
  >
    <form>
      <label
        className="label"
        htmlFor="name"
      >
        Portfolio name
        <input
          className="input"
          disabled={false}
          id="name"
          name="name"
          onChange={[Function]}
          type="text"
          value="Portfolio 1"
        />
      </label>
      <label
        className="label"
        htmlFor="clientName"
      >
        Client name
        <input
          className="input"
          disabled={false}
          id="clientName"
          name="clientName"
          onChange={[Function]}
          type="text"
          value="Kowalski"
        />
      </label>
      <label
        className="label"
        htmlFor="value"
      >
        Initial wallet value
        <input
          className="input"
          disabled={false}
          id="value"
          name="value"
          onChange={[Function]}
          type="text"
          value={100}
        />
      </label>
      <label
        className="label"
        htmlFor="currency"
      >
        Wallet currency
        <input
          className="input"
          disabled={false}
          id="currency"
          name="currency"
          onChange={[Function]}
          type="text"
          value="PLN"
        />
      </label>
      <p
        className="notification is-primary subtitle is-6"
      >
        Please fill all the fields to submit form
      </p>
    </form>
  </div>
</section>
`;

exports[`CreatePortfolio tests should match snapshot with success false 1`] = `
<section
  className="sc-bdVaJa gPOcxX section hero"
>
  <header
    className="title"
  >
    Create portfolio
  </header>
  <div
    className="content is-large"
  >
    <form>
      <div
        className="notification is-warning"
      >
        Creation failed
      </div>
      <label
        className="label"
        htmlFor="name"
      >
        Portfolio name
        <input
          className="input"
          disabled={false}
          id="name"
          name="name"
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <label
        className="label"
        htmlFor="clientName"
      >
        Client name
        <input
          className="input"
          disabled={false}
          id="clientName"
          name="clientName"
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <label
        className="label"
        htmlFor="value"
      >
        Initial wallet value
        <input
          className="input"
          disabled={false}
          id="value"
          name="value"
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <label
        className="label"
        htmlFor="currency"
      >
        Wallet currency
        <input
          className="input"
          disabled={false}
          id="currency"
          name="currency"
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <p
        className="notification is-primary subtitle is-6"
      >
        Please fill all the fields to submit form
      </p>
    </form>
  </div>
</section>
`;

exports[`CreatePortfolio tests should match snapshot with success true 1`] = `
<section
  className="sc-bdVaJa gPOcxX section hero"
>
  <header
    className="title"
  >
    Create portfolio
  </header>
  <div
    className="content is-large"
  >
    <form>
      <div
        className="notification is-success"
      >
        Creation successful
      </div>
      <label
        className="label"
        htmlFor="name"
      >
        Portfolio name
        <input
          className="input"
          disabled={false}
          id="name"
          name="name"
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <label
        className="label"
        htmlFor="clientName"
      >
        Client name
        <input
          className="input"
          disabled={false}
          id="clientName"
          name="clientName"
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <label
        className="label"
        htmlFor="value"
      >
        Initial wallet value
        <input
          className="input"
          disabled={false}
          id="value"
          name="value"
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <label
        className="label"
        htmlFor="currency"
      >
        Wallet currency
        <input
          className="input"
          disabled={false}
          id="currency"
          name="currency"
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <p
        className="notification is-primary subtitle is-6"
      >
        Please fill all the fields to submit form
      </p>
    </form>
  </div>
</section>
`;
